server/src/main/java/com/cloud/server/StatsCollector.java

```diff
@@ -454,8 +456,9 @@ protected void init(Map<String, String> configs) {
             s_logger.warn("Usage stats job aggregation range is to small, using the minimum value of " + UsageUtils.USAGE_AGGREGATION_RANGE_MIN);
             _usageAggregationRange = UsageUtils.USAGE_AGGREGATION_RANGE_MIN;
         }
-        _diskStatsUpdateExecutor.scheduleAtFixedRate(new VmDiskStatsUpdaterTask(), (endDate - System.currentTimeMillis()), (_usageAggregationRange * 60 * 1000),
```

Comment:
Potential overflow in 2 Values before it is converted to long by use in an invocation context.